### PR TITLE
Build: Fix extra config files to work with more than one file

### PR DESCRIPTION
**Pull request by rjernst**
The extra config copy task for integ tests was previously overwriting
the source and destination for each file to be copied, leaving only the
last. This fixes it to create sub copy specs (which is what was
originally intended) for each file.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy [hunk 1]
```diff
@@ -258,9 +259,12 @@ class ClusterFormationTasks {
                 }
             }
             File destConfigFile = new File(node.homeDir, 'config/' + extraConfigFile.getKey())
-            copyConfig.into(destConfigFile.canonicalFile.parentFile)
-                      .from({ extraConfigFile.getValue() }) // wrap in closure to delay resolution to execution time
-                      .rename { destConfigFile.name }
+            // wrap source file in closure to delay resolution to execution time
+            copyConfig.from({ extraConfigFile.getValue() }) {
+                // this must be in a closure so it is only applied to the single file specified in from above
```
Comment (nik9000) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: Move this line down one? The next line isn't in a closure.

**Comment by nik9000** [comment]
Left minor comment. LGTM

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy (hunk 1 above)
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: The line does contain a closure. Notice the curlies inside the parens for `from`. :)

Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: And the other comment about a closure is inside the closure itself.

**Comment by tlrx** [comment]
@rjernst Thanks a lot

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy (hunk 1 above)
Comment (nik9000) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: Ah, yeah - its the trailing `{` on line 263 I missed.

**Pull request merged**